File: src/main/java/com/example/api_immobilier/app/models/ResponseData.java
package com.example.api_immobilier.app.models;

public class ResponseData {
    private String message;
    private boolean status;
    private Object data;
    public ResponseData() {
    }
    public ResponseData(String message, boolean status, Object data) {
        this.message = message;
        this.status = status;
        this.data = data;
    }
    public String getMessage() {
        return message;
    }
    public void setMessage(String message) {
        this.message = message;
    }
    public boolean isStatus() {
        return status;
    }
    public void setStatus(boolean status) {
        this.status = status;
    }
    public Object getData() {
        return data;
    }
    public void setData(Object data) {
        this.data = data;
    }
}

File: app/Models/ResponseData.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

/**
 * @OA\Schema(
 *     title="ResponseData",
 *     description="",
 *     @OA\Xml(
 *         name="ResponseData"
 *     )
 * )
 */
class ResponseData
{

    /**
     * @OA\Property(
     *     title="code"
     * )
     *
     * @var integer
     */
    private $code;

    /**
     * @OA\Property(
     *     title="status"
     * )
     *
     * @var boolean
     */
    private $status;

    /**
     * @OA\Property(
     *     title="message"
     * )
     *
     * @var string
     */
    private $message;

    /**
     * @OA\Property(
     *     title="data"
     * )
     *
     * @var object
     */
    private $data;

}
